kasa/smartprotocol.py

```diff
@@ -101,17 +102,51 @@ async def _query(self, request: Union[str, Dict], retry_count: int = 3) -> Dict:
         # make mypy happy, this should never be reached..
         raise SmartDeviceException("Query reached somehow to unreachable")
 
+    async def _execute_multiple_query(self, request: Dict, retry_count: int) -> Dict:
+        requests = []
+        multi_result = {}
+        smart_method = "multipleRequest"
+        for method, params in request.items():
+            requests.append({"method": method, "params": params})
+
+        end = len(requests)
+        step = (
+            self.MULTI_REQUEST_BATCH_SIZE
+        )  # Break the requests down as there seems to be a size limit
+        for i in range(0, end, step):
+            x = i
+            requests_step = requests[x : x + step]
+
+            smart_params = {"requests": requests_step}
+            smart_request = self.get_smart_request(smart_method, smart_params)
+            _LOGGER.debug(
+                "%s multi-request-batch-%s >> %s",
+                self._host,
+                i + 1,
+                _LOGGER.isEnabledFor(logging.DEBUG) and pf(smart_request),
```

Comment:
`_LOGGER.isEnabledFor(logging.DEBUG)` has a cost as well because it has to check every logger up the chain to see if any of them have debug on.  it would be better to check it once outside the loop

debug_enabled = _LOGGER.isEnabledFor(logging.DEBUG)

Then guard the _LOGGER.debugs